pkg/install/resources.go

```diff
@@ -132,3 +139,81 @@ func VolumeSnapshotLocation(namespace, provider string, config map[string]string
 		},
 	}
 }
+
+func Secret(namespace string, data []byte) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: objectMeta(namespace, "cloud-credentials"),
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		Data: map[string][]byte{
+			"cloud": data,
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+}
+
+func appendUnstructured(list *unstructured.UnstructuredList, obj runtime.Object) error {
+	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&obj)
+	if err != nil {
+		return err
+	}
+	list.Items = append(list.Items, unstructured.Unstructured{Object: u})
+	return nil
+}
+
+// AllResources returns a list of all resources necessary to install Velero, in the appropriate order, into a Kubernetes cluster.
+// Items are unstructured, since there are different data types returned.
+func AllResources(namespace, image, backupStorageProviderName, bucketName, prefix string, bslConfig, vslConfig map[string]string, secretData []byte, restoreOnly bool) (*unstructured.UnstructuredList, error) {
+	resources := new(unstructured.UnstructuredList)
+	// Set the GVK so that the serialization framework outputs the list properly
+	resources.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "List"})
+
+	for _, crd := range CRDs() {
+		unstructuredCRD, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&crd)
+		// Remove the status field, as it's added with null values by the code, but is rejected by server-side validation
+		delete(unstructuredCRD, "status")
+		if err != nil {
+			// TODO: Wrap the error
+			return nil, err
+		}
+		resources.Items = append(resources.Items, unstructured.Unstructured{Object: unstructuredCRD})
+	}
+
+	ns := Namespace(namespace)
+	appendUnstructured(resources, ns)
+
+	crb := ClusterRoleBinding(namespace)
+	appendUnstructured(resources, crb)
+
+	sa := ServiceAccount(namespace)
+	appendUnstructured(resources, sa)
+
+	sec := Secret(namespace, secretData)
+	appendUnstructured(resources, sec)
+
+	bsl := BackupStorageLocation(namespace, backupStorageProviderName, bucketName, prefix, bslConfig)
+	appendUnstructured(resources, bsl)
+
+	vsl := VolumeSnapshotLocation(namespace, backupStorageProviderName, vslConfig)
+	appendUnstructured(resources, vsl)
+
+	deploy := Deployment(namespace,
+		WithImage(image),
+	)
+	if restoreOnly {
+		deploy = Deployment(namespace,
+			WithImage(image),
+			WithRestoreOnly(),
+		)
+	}
+	appendUnstructured(resources, deploy)
+
+	ds := DaemonSet(namespace,
```

Comment:
I think we probably want to add a flag to optionally deploy the restic daemonset, since for folks not using restic it's completely unnecessary and a waste of resources.